e2e-tests-rust/tests/lib.rs

```diff
@@ -651,3 +675,102 @@ async fn dump_state_on_fork() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn load_state_on_run() -> anyhow::Result<()> {
+    let temp_dir = TempDir::new("load-state-test").expect("failed creating temporary dir");
+    let dump_path = temp_dir.path().join("load_state_run.json");
+    let provider = init_testing_provider(identity).await?;
+    let receipts = [
+        provider.tx().finalize().await?,
+        provider.tx().finalize().await?,
+    ];
+    let blocks = provider.get_blocks_by_receipts(&receipts).await?;
+    let state_bytes = provider.anvil_dump_state().await?;
+    drop(provider);
+
+    let mut decoder = GzDecoder::new(&state_bytes.0[..]);
+    let mut json_str = String::new();
+    decoder.read_to_string(&mut json_str).unwrap();
+    let state: VersionedState = serde_json::from_str(&json_str).unwrap();
+    write_json_file(&dump_path, &state)?;
+
+    let dump_path_clone = dump_path.clone();
+    let new_provider = init_testing_provider(move |node| {
+        node.path(get_node_binary_path())
+            .arg("--state-interval")
+            .arg("1")
+            .arg("--load-state")
+            .arg(dump_path_clone.to_str().unwrap())
+    })
+    .await?;
+
+    new_provider.assert_has_receipts(&receipts).await?;
+    new_provider.assert_has_blocks(&blocks).await?;
+    new_provider
+        .assert_balance(receipts[0].sender()?, DEFAULT_TX_VALUE)
+        .await?;
+    new_provider
+        .assert_balance(receipts[1].sender()?, DEFAULT_TX_VALUE)
+        .await?;
+
+    drop(new_provider);
+
+    assert!(
+        dump_path.exists(),
+        "State dump file should still exist at {:?}",
+        dump_path
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn load_state_on_fork() -> anyhow::Result<()> {
+    let temp_dir = TempDir::new("load-state-fork-test").expect("failed creating temporary dir");
+    let dump_path = temp_dir.path().join("load_state_fork.json");
+    let provider = init_testing_provider(identity).await?;
+    let receipts = [
+        provider.tx().finalize().await?,
+        provider.tx().finalize().await?,
+    ];
+    let blocks = provider.get_blocks_by_receipts(&receipts).await?;
+    let state_bytes = provider.anvil_dump_state().await?;
+    drop(provider);
+
+    let mut decoder = GzDecoder::new(&state_bytes.0[..]);
+    let mut json_str = String::new();
+    decoder.read_to_string(&mut json_str).unwrap();
+    let state: VersionedState = serde_json::from_str(&json_str).unwrap();
+    write_json_file(&dump_path, &state)?;
+
+    let dump_path_clone = dump_path.clone();
+    let new_provider = init_testing_provider(move |node| {
+        node.path(get_node_binary_path())
+            .arg("--state-interval")
+            .arg("1")
+            .arg("--load-state")
+            .arg(dump_path_clone.to_str().unwrap())
+            .fork("sepolia-testnet")
+    })
+    .await?;
```

Comment:
@itegulov I've noticed some flakiness with the fork tests timing out during initialization (e.g. test has been running for longer then x). 

Any recommendations to resolve?

Reply:
Yeah I had a feeling this would be an issue. I think we should not do fork tests in this form right now. To minimize flakiness we should set up a real local ZKsync enviornment and use as forking source in all of the fork e2e tests. For example we could spin up dockerized zksync-era node and then seal some blocks.

In any case I think it's for the better to delete these for now.

Reply:
Skipping the fork tests for now, and created issue: #508